MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -874,6 +885,166 @@ namespace WPEFramework {
                     returnResponse(result);
                 }
 
+        /*
+         * @brief This function stops the maintenance activity.
+         * @param1[in]: {"jsonrpc":"2.0","id":"3","method":"org.rdk.MaintenanceManager.1.stopMaintenance",
+         *                  "params":{}}''
+         * @param2[out]:{"jsonrpc":"2.0","id":3,"result":{"success":<bool>}}
+         * @return: Core::<StatusCode>
+         */
+
+        uint32_t MaintenanceManager::stopMaintenance(const JsonObject& parameters,
+                JsonObject& response){
+
+            pid_t pid_num=-1;
+
+            int k_ret=EINVAL;
+            int i=0,record=-1;
+            int32_t exec_status=E_NOK;
+
+            bool task_status[4]={false};
+            bool result=false;
+            bool task_incomplete=false;
+
+            /* only based on RFC */
+            if( checkAbortFlag() ){
+
+                /* run only when the maintenance status is MAINTENANCE_STARTED */
+                m_statusMutex.lock();
+                if ( MAINTENANCE_STARTED == m_notify_status  ){
+
+                    // Set the condition flag m_abort_flag to true
+                    m_abort_flag = true;
+
+                    auto task_status_DCM=m_task_map.find("/lib/rdk/StartDCM_maintaince.sh");
+                    auto task_status_RFC=m_task_map.find(task_names_foreground[0].c_str());
+                    auto task_status_FWDLD=m_task_map.find(task_names_foreground[1].c_str());
+                    auto task_status_LOGUPLD=m_task_map.find(task_names_foreground[2].c_str());
+
+                    task_status[0] = task_status_DCM->second;
+                    task_status[1] = task_status_RFC->second;
+                    task_status[2] = task_status_FWDLD->second;
+                    task_status[3] = task_status_LOGUPLD->second;
+
+                    for (i=0;i<4;i++)
+                        LOGINFO("task status [%d]  = %s ScriptName %s",i,(task_status[i])? "true":"false",script_names[i].c_str());
+                    for (i=0;i<4;i++){
+                        if(task_status[i]){
+                            record = i;
+                            LOGINFO("Checking the Task PID\n");
+                            pid_num=getTaskPID(script_names[i].c_str());
+                            LOGINFO("PID of script_name [%d] = %s is %d \n", i,script_names[i].c_str(),pid_num);
+                            if( pid_num != -1){
+                                /* send the signal to task to terminate */
+                                k_ret=kill(pid_num,SIGABRT);
```

Comment:
Should this be a SIGABRT or SIGTERM? Is this getting trapped by the scripts for graceful exit?

Reply:
SIGTERM is getting ignored in the case SOLICITED maintenance
Due to this issue, we moved to SIGABRT
all scripts have signal handlers and exiting gracefully.

Reply:
SIGABRT handled as part of https://gerrit.teamccp.com/#/q/topic:RDK-33288_2105_sprint+(status:open+OR+status:merged)